coorx/composite.py

```diff
@@ -33,13 +36,26 @@ def __init__(self, *transforms):
             else:
                 trs.append(tr)
         self.transforms = trs
-        for i in range(len(trs)-1):
-            if trs[i].systems[1] != trs[i+1].systems[0]:
-                raise TypeError(f"Coordinate systems of transform {trs[i]} ({trs[i].systems[1]}) does not map to {trs[i+1]} ({trs[i+1].systems[0]})")
+        for i in range(len(trs) - 1):
+            if trs[i].systems[1] != trs[i + 1].systems[0]:
+                raise TypeError(
+                    f"Coordinate systems of transform {trs[i]} ({trs[i].systems[1]}) does not map to {trs[i+1]} ({trs[i+1].systems[0]})"
+                )
 
     @property
     def systems(self):
-        return (self.transforms[0].systems[0], self.transforms[-1].systems[1])
+        return self.transforms[0].systems[0], self.transforms[-1].systems[1]
+
+    def set_systems(self, from_cs, to_cs, cs_graph=None):
+        raise DependentTransformError("Cannot set systems on a CompositeTransform")
+
+    def copy(self, from_cs=None, to_cs=None):
```

Comment:
Is this needed? Seems like you will eventually end up with the same error via set_systems anyway

Reply:
I guess it is; we use dunder-setstate in copy, and Composite overrides that to skip the set_systems call.